library/network/src/modules/SuSEFirewalld.rb

```diff
@@ -0,0 +1,1301 @@
+# encoding: utf-8
+
+# ***************************************************************************
+#
+# Copyright (c) 2016 Novell, Inc.
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of version 2 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail,
+# you may find current contact information at www.novell.com
+#
+# ***************************************************************************
+#
+# Package: SuSEFirewall configuration
+# Summary: Interface manipulation of /etc/sysconfig/SuSEFirewall
+# Authors: Lukas Ocilka <locilka@suse.cz
+#
+# $Id$
+#
+# Module for handling SuSEfirewall2
+# DO NOT use this class directly. Use SuSEFirewallClass in your modules
+require "yast"
+require "network/firewalld"
+require_relative "SuSEFirewall"
+
+module Yast
+  # ----------------------------------------------------------------------------
+  # SuSEFirewalld Class. Trying to provide relevent pieces of SF2 functionality via
+  # firewalld.
+  class SuSEFirewalldClass < SuSEFirewallClass
+    include Firewalld
+    require "set"
+    attr_reader :special_all_interface_zone
+
+    # Valid attributes for firewalld zones
+    # :interfaces = [Array<String>]
+    # :masquerade = Boolean
+    # :modified   = [Set<Symbols>]
+    # :ports      = [Array<String>]
+    # :protocols  = [Array<String>]
+    # :services   = [Array<String>]
+    @@zone_attributes = [:interfaces, :masquerade, :modified, :ports, :protocols, :services]
+    # {enable,start}_firewall are "inherited" from SF2 so we can't use symbols
+    # there without having to change all the SF2 callers.
+    @@key_settings = ["enable_firewall", "logging", "routing", "start_firewall"]
+
+    EMPTY_ZONE = {
+      interfaces: [],
+      masquerade: false,
+      modified:   Set.new,
+      ports:      [],
+      protocols:  [],
+      services:   []
+    }
+
+    # We need that for the tests. Nothing else should access the API
+    # directly
```

Comment:
so here it make sense to use yardoc @api call. Btw if you need to access it from tests only you can use http://apidock.com/ruby/Object/instance_variable_get which do not need public access ( but use it only in tests, not in production code as it break encapsulation )